app/build.gradle.kts

```diff
@@ -109,15 +109,15 @@ android {
 
 dependencies {
     // Development
-    implementation("androidx.activity:activity-compose:1.8.2")
-    implementation(platform("androidx.compose:compose-bom:2024.04.00"))
+    implementation("androidx.activity:activity-compose:1.9.0")
+    implementation(platform("androidx.compose:compose-bom:2024.04.01"))
     implementation("androidx.compose.ui:ui")
     implementation("androidx.compose.ui:ui-graphics")
     implementation("androidx.compose.ui:ui-tooling-preview")
     implementation("androidx.compose.material3:material3")
     implementation("androidx.compose.material:material-icons-extended")
-    implementation("androidx.core:core-ktx:1.12.0")
-    implementation("androidx.datastore:datastore-preferences-core:1.0.0")
+    implementation("androidx.core:core-ktx:1.13.0")
+    implementation("androidx.datastore:datastore-preferences-core:1.0.0") // do NOT update this
```

Comment:
Instead of this comment, put something more informative: 1.0.1 version contains bugs and breaks stuff. Evade it.